Sources/Views/OATargetPointView.mm

```diff
@@ -1998,16 +1998,28 @@ - (IBAction) buttonShareClicked:(id)sender
 {
     // http://osmand.net/go.html?lat=12.6313&lon=-7.9955&z=8&title=New+York The location was shared with you by OsmAnd
     
-    UIImage *image = [self.mapView getGLScreenshot];
+    UIActivityViewController *activityViewController;
     
-    //NSString *title = [_targetPoint.title stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet alphanumericCharacterSet]];
-    
-    NSString *string = [NSString stringWithFormat:kShareLinkTemplate, _targetPoint.location.latitude, _targetPoint.location.longitude, _mapView.zoomLevel];
-    
-    UIActivityViewController *activityViewController =
-    [[UIActivityViewController alloc] initWithActivityItems:@[image, string]
-                                      applicationActivities:nil];
+    if (_previousTargetType == OATargetFavorite) {
+        NSString *title = [self prepareForSharing: _targetPoint.title];
+        NSString *group = [self prepareForSharing: _customController.getTypeStr];
```

Comment:
title, group and description have taken from totally different places. Use _targetPoint or _customController or OAFavoriteViewController (but not OAEditTargetViewController since we know that this is OATargetFavorite)